### DISCUSSION TITLE
Attempted to read or write protected memory. This is often an indication that other memory is corrupt

### DISCUSSION THREAD

**Discussion by trancaodua** [cefsharp/CefSharp#3779] [General]
-I use CefSharp.WinForms  for my application. When I run the app, it sometimes crashes
System.AccessViolationException
  HResult=0x80004003
  Message=Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
  Source=mscorlib
  StackTrace:
   at System.String.wcslen(Char* ptr)
   at System.String.CtorCharPtr(Char* ptr)
   at CefSharp.Internals.CefStringVisitorAdapter.Visit(CefStringVisitorAdapter* , CefStringBase<CefStringTraitsUTF16>* string)

  This exception was originally thrown at this call stack:
    [External Code]

-Version: 92.0.26
-Visual studio: 2019
-Net framework: 4.7.2
-Target build: x64
![image](https://user-images.githubusercontent.com/62211391/131216002-c0488a07-2a6a-44c4-98d8-e8c7ea0d9c5e.png)

**Comment by amaitland** [comment]
Please provide a detailed code example.

**Reply by amaitland** [reply]
I've added a speculative fix which will be included in the final `93` release (not in the `-pre` release). Commit https://github.com/cefsharp/CefSharp/commit/880c4662676b4a1ea680803f277e5c50d32e4839

Without an actual example and only a partial stack trace I'm taking an educated guess here.

**Reply by trancaodua** [reply]
thanks you very much
